sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -606,6 +608,22 @@ func withPersistentStorage(containerID string) oci.SpecOpts {
 	return oci.Compose(oci.WithMounts(persistentMounts))
 }
 
+// withProxyEnv reads proxy environment variables and returns a spec option for passing said proxy environment variables
+func withProxyEnv() oci.SpecOpts {
+	noOp := func(_ context.Context, _ oci.Client, _ *containers.Container, s *runtimespec.Spec) error { return nil }
+	httpsProxy, httpsProxySet := os.LookupEnv("HTTPS_PROXY")
+	noProxy, noProxySet := os.LookupEnv("NO_PROXY")
+	withHttpsProxy := noOp
+	withNoProxy := noOp
+	if httpsProxySet {
+		withHttpsProxy = oci.WithEnv([]string{"HTTPS_PROXY=" + httpsProxy, "https_proxy=" + httpsProxy})
+	}
+	if noProxySet {
+		withNoProxy = oci.WithEnv([]string{"NO_PROXY=" + noProxy, "no_proxy=" + noProxy})
+	}
+	return oci.Compose(withHttpsProxy, withNoProxy)
+}
```

Comment:
Do we need to set empty environment variables?  If we can skip that, `os.Getenv` ends up being a bit simpler.  And we can rewrite this in a slightly simpler way too.

```suggestion
	withHttpsProxy := noOp
	if val := os.Getenv("HTTPS_PROXY"); val != "" {
		withHttpsProxy = oci.WithEnv([]string{
			"HTTPS_PROXY=" + val,
			"https_proxy=" + val,
		})
	}
	withNoProxy := noOp
	if val := os.Getenv("NO_PROXY"); val != "" {
		withNoProxy = oci.WithEnv([]string{
			"NO_PROXY=" + val,
			"no_proxy=" + val,
		})
	}
	return oci.Compose(withHttpsProxy, withNoProxy)
}
```

Reply:
I think we want to differentiate between the variables being unset and being null just in case there are applications that care about that distinction. (See https://github.com/bottlerocket-os/bottlerocket/pull/1432#discussion_r604511299)

Reply:
That comment is about "set but empty" environment variables.  The approach I outlined above avoids that (checking for `val != ""`) and also avoids propagating "set but empty" environment variables into the host containers.  If we think "set but empty" variables _should_ be propagated, then we need `os.LookupEnv`.

Reply:
> If we think "set but empty" variables should be propagated, then we need os.LookupEnv.

I understood the ask as to _not_ propagate the environment variable if it was unset in the first place, but propagate it if it was set regardless of whether it's empty or not, thus the `os.LookupEnv`.

Can you please clarify, @bcressey?

Reply:
It seems 'safer' to mirror the vars exactly (e.g. set-but-empty if that's what we find) rather than to assume we know better.